compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -860,7 +860,7 @@ trait Implicits:
   /** Find an implicit conversion to apply to given tree `from` so that the
    *  result is compatible with type `to`.
    */
-  def inferView(from: Tree, to: Type)(using Context): SearchResult = {
+  def inferView(from: Tree, to: Type, maybeNameSpan: Option[Span] = None)(using Context): SearchResult = {
```

Comment:
We already a `NoSpan` "null" object we can use, instead of having to wrap it with Option.  Could you switch to that, please?  You can then call `.exists` on the span (which just compares to NoSpan) instead of calling `foldLeft` (which also creates another lambda).